src/storage-types/src/stats.rs

```diff
@@ -174,6 +174,24 @@ fn downcast_stats<'a, T: Data>(
 }
 
 impl RelationPartStats<'_> {
+    pub fn may_match_mfp<'a>(&'a self, time_range: ResultSpec<'a>, mfp: &MapFilterProject) -> bool {
```

Comment:
Ideally this would be called from within `persist_source` as well, but that has an `MfpPlan` and not a `MapFilterProject`. IMO it would make sense to pass in the latter instead, but I'd like to leave that to a future PR to limit the blast radius.